### PR TITLE
Added async performance test

### PR DESCRIPTION
**Pull request by shinglyu**
Add a new way to test arbitrary timing from JavaScript (only for Gecko). This is for Stylo testing.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [x] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15067)

### CODE REVIEW

**Comment by shinglyu** [comment]
Hmmm, I rebased the the merge conflict are not resolved, any suggestions?

**Comment by shinglyu** [comment]
Oh, resolved.

**Comment by jdm** [comment]
@Manishearth This is waiting on your review.

File: etc/ci/performance/harness/harness.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+window.customTimers = {};
+// Create a custome timestamp with a custom name
+function mark(name) {
+  if (window.performance) {
+    // performance.now() is the time after navigationStart
+    // https://developer.mozilla.org/zh-TW/docs/Web/API/Performance/now
```
Comment (Manishearth) [inline etc/ci/performance/harness/harness.js]: en-US? Not too important I guess.

**Review by Manishearth — APPROVED** [review]
r=me

File: etc/ci/performance/harness/harness.js (hunk 1 above)
Comment (KiChjang) [inline etc/ci/performance/harness/harness.js]: 'MURRICA

**Pull request merged**